# tugas - 3 : K-Nearest Neighbors 

Link colab tugas 3 :


https://colab.research.google.com/drive/1nWbHSZeU4cQ_kiptRjSxPi8erDGTeV7a#scrollTo=yHEcOECkGVuY

#### Persiapan Lingkungan


*   Impor *train_test_split* dari *sklearn*
*   Impor *KNeighborsClassifier* dari *sklearn*
*   Impor *pandas library*



In [2]:
from sklearn.model_selection import train_test_split#split dataset into train and test data

In [3]:
from sklearn.neighbors import KNeighborsClassifier# Create KNN classifier

In [4]:
import pandas as pd

In [5]:
dataset_url = "https://raw.githubusercontent.com/Iqbalsuryaa/datamining/main/Iris.csv"

In [6]:
df = pd.read_csv(dataset_url)
df.head()

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm      Species
0   1            5.1           3.5            1.4           0.2  Iris-setosa
1   2            4.9           3.0            1.4           0.2  Iris-setosa
2   3            4.7           3.2            1.3           0.2  Iris-setosa
3   4            4.6           3.1            1.5           0.2  Iris-setosa
4   5            5.0           3.6            1.4           0.2  Iris-setosa

In [7]:
#check number of rows and columns in dataset
df.shape

(150, 6)

### Hitung Data

*   Pisahkan kolom "variety" dari *dataframe*
*   Ambil kolom "variety" sebagai target kolom untuk kategori kelas
*   Pisahkan data latih dengan data tes


```
# data latih (nilai data)
X_train 

# data tes (nilai data)
X_test 

# data latih (kelas data)
y_train

# data tes (kelas data)
y_test
```





In [9]:
#create a dataframe with all training data except the target column
X = df.drop(columns=["Species"])#check that the target variable has been removed
X.head()

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm
0   1            5.1           3.5            1.4           0.2
1   2            4.9           3.0            1.4           0.2
2   3            4.7           3.2            1.3           0.2
3   4            4.6           3.1            1.5           0.2
4   5            5.0           3.6            1.4           0.2

In [10]:
# separate target values
y = df["Species"].values  # view target values
y[0:5]

array(['Iris-setosa', 'Iris-setosa', 'Iris-setosa', 'Iris-setosa',
       'Iris-setosa'], dtype=object)

In [11]:
# percent_amount_of_test_data = / HUNDRED_PERCENT
percent_amount_of_test_data = 0.2

In [12]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = percent_amount_of_test_data, random_state=1, stratify=y)



*   Tentukan nilai K = 10 (10 tetangga terdekat)
*   Buat data latih


```
knn_classifier.fit(train_data)
```




In [13]:
amount_of_neighbor = 3

In [14]:
# Fit the classifier to the data
knn = KNeighborsClassifier(n_neighbors = amount_of_neighbor)

# create train data
knn.fit(X_train,y_train)

KNeighborsClassifier(n_neighbors=3)



*   Tes data target terhadap data latih
*   Tampilkan predksi 10 data pertama



```
knn_classifier.predict(test_data)
```





In [15]:
#show first 5 model predictions on the test data
knn.predict(X_test)[0:30]

array(['Iris-virginica', 'Iris-setosa', 'Iris-versicolor', 'Iris-setosa',
       'Iris-setosa', 'Iris-setosa', 'Iris-virginica', 'Iris-virginica',
       'Iris-virginica', 'Iris-versicolor', 'Iris-setosa',
       'Iris-versicolor', 'Iris-virginica', 'Iris-versicolor',
       'Iris-virginica', 'Iris-setosa', 'Iris-virginica',
       'Iris-versicolor', 'Iris-versicolor', 'Iris-virginica',
       'Iris-versicolor', 'Iris-versicolor', 'Iris-setosa', 'Iris-setosa',
       'Iris-virginica', 'Iris-virginica', 'Iris-setosa', 'Iris-setosa',
       'Iris-versicolor', 'Iris-versicolor'], dtype=object)



*   Cetak prediksi nilai dari data tes



```
knn_classifier.score(test_data)
```





In [16]:
#check accuracy of our model on the test data
knn.score(X_test, y_test)

1.0

### Kesimpulan


*   Dataset bunga *iris* terdiri dari 150 data
*   Ambil 80% dari *iris dataset* sebagai data latih
*   Ambil 20% dari *iris dataset* sebagai data tes
*   Tetapkan nilai K = 10
*   Hasil 10 data pertama dari 20% data latih


```
['Virginica', 'Setosa', 'Versicolor', 'Setosa', 'Setosa', 'Setosa',
       'Virginica', 'Virginica', 'Virginica', 'Versicolor']
```
*   Dengan akurasi data yaitu $ 0.967\% $



